Release docs: v2.8.0 (#678)
CERT-Polska/mwdb-core: docs/rich-attributes-guide.rst

Rich attributes guide
=====================

.. versionadded:: 2.8.0

.. warning::
    This feature is still **in development** and it may significantly change in the future versions.

Attributes can be used to store complex information in JSON format. In addition, values can be queried just like

> configurations using ``attribute.<key>:<value>`` syntax.

Reviewer: ```suggestion
configurations with the use of ``attribute.<key>:<value>`` syntax.
```

> Rich attribute templates combine forces of Mustache and Markdown languages to give you next level of flexibility in rendering

Reviewer: ```suggestion
Rich attribute templates combine forces of Mustache and Markdown languages that gives you next level of flexibility in rendering
```

attribute values in UI without the need of writing additional plugins. Templates allow you to render value objects as custom
links, lists, tables or combinations of them. You can also combine your representation with other context values like
sample name or hash.

Rich attributes enable us to produce similar reports like Details section in VirusTotal
(e.g. https://www.virustotal.com/gui/file/32fae9922417d6405bf60144d819d5e02b44060fa8f07e5e71c824725f44307f/details)
based on data contained in attribute objects. All templates are rendered client-side.

Getting started
---------------

Go to ``Settings`` -> ``Attributes`` and choose attribute key for edit.

.. image:: ./_static/rich-template-attribute-details.png
   :target: ./_static/rich-template-attribute-details.png
   :alt: Attribute details

Then click on ``Edit`` button next to the ``Rich template`` field to open the interactive template editor.

The view consists of two edit fields. The first one is **Template** that can be filled with Mustache&Markdown
template used for rendering attribute value. The second one is **Example value** that contains example attribute value
appropriate for attribute key in JSON format.

> **Preview** field below shows the rendered representation based on template and example value.

Reviewer: ```suggestion
**Preview** field shows the rendered representation based on template and example value.
```

.. image:: ./_static/rich-template-edit.png
   :target: ./_static/rich-template-edit.png
   :alt: Rich template edit view

> After filling fields with appropriate template, use ``Store`` to save the template or ``Cancel`` to leave it unedited.

Reviewer: ```suggestion
After filling fields with the appropriate template, use ``Store`` to save the template or ``Cancel`` to leave it unchanged.
```

If you want to clear template and show only raw representation of attribute value, click on ``Delete``.

> Interactive editor contains few sample templates that can be used for making your own definitions.

Reviewer: Depends on meaning
```suggestion
Interactive editor contains few sample templates that can be used for making its own definitions.
```

Mustache basics
---------------

Mustache is basic logic-less template language. Mustache substitution is the first stage of template processing.

The most basic Mustache tag is ``{{value}}`` that will be substituted by Markdown-escaped attribute value.

> If value is represented by JSON object, we can reference the appropriate subkey using dot as a key separator.

Reviewer: ```suggestion
If value is represented by JSON object, we can reference the appropriate subkey with the use of dot as a key separator.
```
